src/win32_window.c

```diff
@@ -893,47 +898,58 @@ static int createNativeWindow(_GLFWwindow* window,
     DWORD style = getWindowStyle(window);
     DWORD exStyle = getWindowExStyle(window);
 
-    if (window->monitor)
-    {
-        GLFWvidmode mode;
-
-        // NOTE: This window placement is temporary and approximate, as the
-        //       correct position and size cannot be known until the monitor
-        //       video mode has been picked in _glfwSetVideoModeWin32
-        _glfwPlatformGetMonitorPos(window->monitor, &xpos, &ypos);
-        _glfwPlatformGetVideoMode(window->monitor, &mode);
-        fullWidth  = mode.width;
-        fullHeight = mode.height;
-    }
-    else
-    {
-        xpos = CW_USEDEFAULT;
-        ypos = CW_USEDEFAULT;
-
-        if (wndconfig->maximized)
-            style |= WS_MAXIMIZE;
-
-        getFullWindowSize(style, exStyle,
-                          wndconfig->width, wndconfig->height,
-                          &fullWidth, &fullHeight);
-    }
-
-    wideTitle = _glfwCreateWideStringFromUTF8Win32(wndconfig->title);
-    if (!wideTitle)
-        return GLFW_FALSE;
-
-    window->win32.handle = CreateWindowExW(exStyle,
-                                           _GLFW_WNDCLASSNAME,
-                                           wideTitle,
-                                           style,
-                                           xpos, ypos,
-                                           fullWidth, fullHeight,
-                                           NULL, // No parent window
-                                           NULL, // No window menu
-                                           GetModuleHandleW(NULL),
-                                           NULL);
-
-    free(wideTitle);
+	const auto parent = (wndconfig->nativeParent ? wndconfig->nativeParent : NULL);
+
+	if (!parent)
+	{
+		if (window->monitor)
+		{
+			GLFWvidmode mode;
+
+			// NOTE: This window placement is temporary and approximate, as the
+			//       correct position and size cannot be known until the monitor
+			//       video mode has been picked in _glfwSetVideoModeWin32
+			_glfwPlatformGetMonitorPos(window->monitor, &xpos, &ypos);
+			_glfwPlatformGetVideoMode(window->monitor, &mode);
+			fullWidth = mode.width;
+			fullHeight = mode.height;
+		}
+		else
+		{
+			xpos = CW_USEDEFAULT;
+			ypos = CW_USEDEFAULT;
+
+			if (wndconfig->maximized)
+				style |= WS_MAXIMIZE;
+
+			getFullWindowSize(style, exStyle,
+				wndconfig->width, wndconfig->height,
+				&fullWidth, &fullHeight);
+		}
+
+		wideTitle = _glfwCreateWideStringFromUTF8Win32(wndconfig->title);
+		if (!wideTitle)
+			return GLFW_FALSE;
+
+
+
+		window->win32.handle = CreateWindowExW(exStyle,
+			_GLFW_WNDCLASSNAME,
+			wideTitle,
+			style,
+			xpos, ypos,
+			fullWidth, fullHeight,
+			NULL,
+			NULL, // No window menu
+			GetModuleHandleW(NULL),
+			NULL);
+
+		free(wideTitle);
+	}
+	else
+	{
+		window->win32.handle = parent;
+	}
```

Comment:
I am not sure how much upstream glfw cares about this sort of thing, but it would be nice to restructure this to minimize the diff.  Maybe factor out all the existing create code to a different function so the indentation level is the same, then something like

```c
if (parent)
{
    window->win32.handle = parent;
}
else
{
    window->win32.handle = createTopLevelWindow(...);
}
```

However, I'm not sure about the approach here anyway.  I think this should create a new window passing the given parent (which is why it's called "parent"), rather than not create a window at all.  The existing code I know of that does this works this way, and it is how the X11 support works.

I'll try to write a test and experiment it if I get some time.